### DISCUSSION TITLE
Connection between digital simulation and real robot

### DISCUSSION THREAD

**Discussion by petrasvestartas** [visose/Robots#111] [Q&A] [answered]
Hi,

I would like to ask if there are any manuals for this scenario:

a) user buys 2nd hand abb robot and gets delivered to his/her workspace
b) user installs Robots plugin on rhino grasshopper

Since a company that installs robots does not know anything about Robots plugin nor Rhino how a user can establish a physical link between robot and rhino? Is there any proper manuals?

**Comment by robin-gdwl** [comment]
Did you see the section for ABB robots in the wiki? 

```markdown
# abb
In the latest version you can use the Remote component directly from Grasshopper to upload, load and optionally run the program on the controller. You'll need to install the Robot Communication Runtime if you don't have RobotStudio installed.
```

  https://github.com/visose/Robots/wiki/Home 

Is this what you are looking for or is it rather the actual connection (setting up a network, connecting ethernet cables ...) that you need help with?

**Comment by petrasvestartas** [comment]
Dear @robin-gdwl 

Thank you for reply.

The second - rather the actual connection (setting up a network, connecting ethernet cables ...) that you need help with.

**Comment by robin-gdwl** [comment, marked as answer]
I see, I have not yet used ABB Robots so I cant help you  myself. 
It probably also depends on what robot and controller you use. 
This video should be a good starting point: (dont let the UR-Robot confuse you its about connecting to ABB)
https://www.youtube.com/watch?v=LiV7xgSIZKA 
Since this plugin (Robots) doesnt connect directly to the robot but instead uses Robot Studio or [ABB Robot communication Runtime ](https://forums.robotstudio.com/discussion/10677/robot-communication-runtime-6-06-01-sp1-is-available-for-download) to interface with the robot I would check if you can find a manual for those that details how to connect. 
Abb has rather [comprehensive documentation open for download](https://new.abb.com/products/robotics/robotstudio/downloads) so it should be possible to find the information you need. 

As per the [Robotstudio Manual](https://search.abb.com/library/Download.aspx?DocumentID=3HAC032104-001&LanguageCode=en&DocumentPartId=&Action=Launch) on page 19 you can find the system requirements and necessary settings. 

![image](https://user-images.githubusercontent.com/25942644/114183602-22774600-9944-11eb-8da4-f16a1121deaa.png)

And starting from pagee 28 you will find a guide on how to connect the robot controller to the PC  

![image](https://user-images.githubusercontent.com/25942644/114184009-987bad00-9944-11eb-93a1-cf3ce4446d9f.png)

**Comment by petrasvestartas** [comment]
Thank @robin-gdwl 

Is it also valid for both old and newer abb i.e. IRB 6400?

**Comment by robin-gdwl** [comment]
> Is it also valid for both old and newer abb i.e. IRB 6400?

I'm not sure, I have only used robotstudio as a simulation not with a physical robot, but I guess this info should be found somewhere in the manuals for roboguide or the controller itself.

**Comment by petrasvestartas** [comment]
Thanks for the help;)

**Comment by visose** [comment]
@robin-gdwl thanks for answering 👍🏻
